PSCompatibilityCollector/Microsoft.PowerShell.CrossCompatibility/Commands/GetPSCompatibilityPlatformDataCommand.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Management.Automation;
+using Microsoft.PowerShell.CrossCompatibility.Collection;
+using Microsoft.PowerShell.CrossCompatibility.Data;
+
+namespace Microsoft.PowerShell.CrossCompatibility.Commands
+{
+    /// <summary>
+    /// Class defining the Get-PSCompatibilityPlatformData cmdlet.
+    /// Assembles an object describing the current platform PowerShell is running on.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, CommandUtilities.MODULE_PREFIX + "PlatformData")]
+    [OutputType(typeof(PlatformData))]
+    public class GetPSCompatibilityPlatformDataCommand : Cmdlet
+    {
+        /// <summary>
+        /// The PowerShell data object to use. If not set, this is generated.
+        /// </summary>
+        [Parameter]
+        public PowerShellData PowerShell { get; set; }
+
+        /// <summary>
+        /// The operating system data object to use. If not set, this is generated. 
+        /// </summary>
+        [Parameter]
+        public OperatingSystemData OperatingSystem { get; set; }
+
+        /// <summary>
+        /// The .NET data object to use. If not set, this is generated.
+        /// </summary>
+        [Parameter]
+        public DotnetData DotNet { get; set; }
+
+        protected override void EndProcessing()
```

Comment:
do you see a scenario where this would be called iteratively? If so, it should probably be in `ProcessRecord`

Reply:
I thought about that, but because there are no pipeline parameters and it's probably used for collecting a profile on the local machine, `EndProcessing` seemed like the right call. This is effectively a function rather than a filter